src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -167,12 +170,14 @@ BeginTransactionOnShardPlacements(uint64 shardId, char *userName)
 
 		MemoryContextSwitchTo(oldContext);
 
-		/* now that connection is tracked, issue BEGIN */
-		result = PQexec(connection, "BEGIN");
-		if (PQresultStatus(result) != PGRES_COMMAND_OK)
-		{
-			ReraiseRemoteError(connection, result);
-		}
+		/* issue BEGIN */
+		RemoteTransactionBegin(connection);
+
+		/*
+		 * Every individual failure should cause entire distributed
+		 * transaction to fail.
+		 */
+		MarkRemoteTransactionCritical(connection);
```

Comment:
> Shouldn't this be set before the begin?
It should be set before the begin, wouldn't make much of a difference though, we'll fail just a tiny bit later. Will change.

> Also, what if we call this automatically using something like a TRANSACTION_CRITICAL flag?

I wondered about that, but decided that it's not that unlikely that a) the need to set the flag only arises after having interactions with the remote b) it's not known to the connection establishing layer that it needs to be set.

So I'm slightly inclined to leave this as-is.